lib/utils/mongoclient/src/index.js

```diff
@@ -104,19 +105,157 @@ const key = (uri) => {
   return undefined;
 };
 
+// Get uri schema
+const getSchema = (uri) => {
+  const i = uri.indexOf('://');
+  return i != -1 ? uri.substring(0, i) : undefined;
+};
+
+// Get server partitioning mode, either 'coll' or 'db' depending on whether
+// the server uri contains a database or not
+const getPartitioningMode = (server) => {
+  // Remove uri schema and trailing '/'
+  const schema = getSchema(server);
+  let uri = schema ? server.substring(schema.length + 3) : server;
+  uri = uri[uri.length - 1] === '/' ? uri.substring(0, uri.length - 1) : uri;
+
+  // Return 'coll' if the resulting uri still contains '/', otherwise 'db'
+  return uri.includes('/') ? 'coll' : 'db';
```

Comment:
mongo 3.4 adds collation and just `coll` can be a bit misleading. `collection` would be a better name imho